drivers/gles3/storage/texture_storage.cpp

```diff
@@ -808,8 +817,57 @@ void TextureStorage::texture_2d_layered_initialize(RID p_texture, const Vector<R
 	}
 }
 
-void TextureStorage::texture_3d_initialize(RID p_texture, Image::Format, int p_width, int p_height, int p_depth, bool p_mipmaps, const Vector<Ref<Image>> &p_data) {
-	texture_owner.initialize_rid(p_texture, Texture());
+void TextureStorage::texture_3d_initialize(RID p_texture, Image::Format p_format, int p_width, int p_height, int p_depth, bool p_mipmaps, const Vector<Ref<Image>> &p_data) {
+	ERR_FAIL_COND(p_data.size() == 0);
+
+	Image::Image3DValidateError verr = Image::validate_3d_image(p_format, p_width, p_height, p_depth, p_mipmaps, p_data);
+	if (verr != Image::VALIDATE_3D_OK) {
+		ERR_FAIL_MSG(Image::get_3d_image_validation_error_text(verr));
+	}
+
+	Ref<Image> image = p_data[0];
+	{
+		int valid_width = 0;
+		int valid_height = 0;
+		bool valid_mipmaps = false;
+		Image::Format valid_format = Image::FORMAT_MAX;
+
+		for (int i = 0; i < p_data.size(); i++) {
+			ERR_FAIL_COND(p_data[i]->is_empty());
+
+			if (i == 0) {
+				valid_width = p_data[i]->get_width();
+				valid_height = p_data[i]->get_height();
+				valid_format = p_data[i]->get_format();
+				valid_mipmaps = p_data[i]->has_mipmaps();
+			} else {
+				ERR_FAIL_COND(p_data[i]->get_width() != valid_width);
+				ERR_FAIL_COND(p_data[i]->get_height() != valid_height);
+				ERR_FAIL_COND(p_data[i]->get_format() != valid_format);
+				ERR_FAIL_COND(p_data[i]->has_mipmaps() != valid_mipmaps);
+			}
+		}
+	}
```

Comment:
This should logic should mirror the logic from ``RD::texture_3d_initialize()`` instead of the logic from ``GLES3::texture_layered_initialize()`` as 3D textures work differently. 

Notably, 3D textures store mipmaps as slices in p_data, so this code will fail with any 3D texture that uses mipmaps. By way of illustration ``p_data`` is organized as follows (S stands for slice, M stands for mipmap):
```
// A 4x4x4 3D texture without mipmaps would look like:
p_data = {S1M1, S2M1, S3M1, S4M1}

// A 4x4x4 3D texture with mipmaps would look like:
p_data = {S1M1, S2M1, S3M1, S4M1, S1M2, S2M2, S3M2, S4M2, S1M3, S2M3, S3M3, S4M3}
```

https://github.com/godotengine/godot/blob/eca6f0eb545166708c56633152974f40f51de2be/servers/rendering/renderer_rd/storage_rd/texture_storage.cpp#L979-L1022

Reply:
I see, I'll have to slightly refactor it...